src/main/java/it/gov/pagopa/reward/controller/RewardTrxSynchronousApiControllerImpl.java

```diff
@@ -22,12 +24,13 @@ public RewardTrxSynchronousApiControllerImpl(RewardTrxSynchronousApiService rewa
     }
 
     @Override
-    public Mono<SynchronousTransactionResponseDTO> previewTransaction(SynchronousTransactionRequestDTO trxPreviewRequest, String initiativeId) {
+    public Mono<ResponseEntity<SynchronousTransactionResponseDTO>> previewTransaction(SynchronousTransactionRequestDTO trxPreviewRequest, String initiativeId) {
         log.info("[SYNC_PREVIEW_TRANSACTION] The user {} requests preview of a transaction having id {} on initiativeId {}", trxPreviewRequest.getUserId(), trxPreviewRequest.getTransactionId(), initiativeId);
 
         return PerformanceLogger.logTimingFinally("SYNC_PREVIEW_TRANSACTION",
                 rewardTrxSynchronousService.previewTransaction(trxPreviewRequest, initiativeId)
                         .doOnNext(r -> log.info("[SYNC_PREVIEW_TRANSACTION] The preview requested by userId {} of a transaction having id {} on initiativeId {} resulted into status {} and rejections {}", trxPreviewRequest.getUserId(), trxPreviewRequest.getTransactionId(), initiativeId, r.getStatus(), r.getRejectionReasons()))
+                        .map(r -> ResponseEntity.ok().header(HttpHeaders.ETAG, String.valueOf(r.getReward() != null ? r.getReward().getCounters().getVersion() : null)).body(r))
```

Comment:
format better this line in order to make it more readable
```suggestion
                        .map(r -> ResponseEntity
                          .ok()
                          .header(HttpHeaders.ETAG, String.valueOf(r.getReward() != null ? r.getReward().getCounters().getVersion() : null))
                          .body(r))
```